auth/internal/internal.go

```diff
@@ -111,3 +112,9 @@ func GetProjectID(b []byte, override string) string {
 	}
 	return v.Project
 }
+
+// ReadAll consumes the whole reader and safely reads the content of its body
+// with some overflow protection.
+func ReadAll(r io.Reader) ([]byte, error) {
+	return io.ReadAll(io.LimitReader(r, 1<<20))
```

Comment:
Maybe extract this value to: `const maxBodySize = 1 << 20` for clarity?

Reply:
done.